extensions/amp-ad-network-adsense-impl/0.1/amp-ad-network-adsense-impl.js

```diff
@@ -200,8 +201,12 @@ export class AmpAdNetworkAdsenseImpl extends AmpA4A {
 
   /** @override */
   getAdUrl() {
-    // TODO: Check for required and allowed parameters. Probably use
-    // validateData, from 3p/3p/js, after moving it someplace common.
+    const width = Number(this.element.getAttribute('width'));
+    const height = Number(this.element.getAttribute('height'));
+
+    // TODO: check mandatory fields
+    validateAdSenseData(this.element.dataset, height);
```

Comment:
I don't think we should call 3p/validateData as part of Fast Fetch as its mostly related to delayed fetch implementation as adds no real value while introducing a dependency that we'd prefer it not.  Instead, can you only check the responsive height requirement and do it within #isValidElement